### ISSUE TITLE
Allow users to send email to a single user from the user chip, or the profile page.

### ISSUE THREAD

**Issue by roberthouse54** [mitodl/micromasters#2196]
This issue involves allowing users (students and admin) to send messages to other single users via a link from the user chip and the profile page. In both cases we will open a "Send a Message" modal. 

Since this will be the 3rd e-mail UI added to the react app, some refactoring may be desirable -- see #2468 

*** acceptance criteria
- [ ] Add direct email link to the profile page (will replace the user's displayed email address)
- [ ] Restrict this feature to staff and paid learners. ~~For learners who haven't paid, we will display an upsell message~~ _(this upsell message will be implemented in a separate issue)_

![image](https://cloud.githubusercontent.com/assets/20047260/21431928/5b1816ac-c837-11e6-9d5b-059c112a2fdd.png)

![image](https://cloud.githubusercontent.com/assets/20047260/22706559/bfb1f756-ed3d-11e6-8fab-b666bbdcd893.png)

![image](https://cloud.githubusercontent.com/assets/20047260/22706196/5a2b7836-ed3c-11e6-855b-8686811a6832.png)

![image](https://cloud.githubusercontent.com/assets/20047260/22706439/3e4c9130-ed3d-11e6-9677-bb13effcf284.png)

**Comment by gsidebo** [comment]
@pdpinch @roberthouse54 A few important design questions here:

1. Do we currently have some way for learners to actually find other learners? I don't see one
1. The design for the user profile page looks different from what we actually have implemented. Is there supposed to be some design work done before this?
1. One thing we talked about as a benefit of implementing one-way learner-to-learner email in general was to avoid exposing learners' emails to each other. Is that something we're still interested in? If so, we should probably get rid of the email address display when we add the link.

Basically I see 2 places that we can actually implement this right now: (1) on user profile pages, which only staff can reasonably find at the moment, and (2) the user chip on the search results page, which is also staff-only. Is that correct? If so, we should reflect that in the issue description

**Comment by roberthouse54** [comment]
1. We do not currently have this ability for learners to find other learners, but it has been our plan to do it.
2. As to differences in the profile---- do not make changes to the profile that are not required for this feature.
3. Yes, I see your point. Let's not display the email.

**Comment by pdpinch** [comment]
This is blocked by #2641

**Comment by gsidebo** [comment]
@pdpinch just realized that we have an odd edge case. the issue description says that only paid users should be able to message other users. let's say that the sender is enrolled in programs A and B, and is unpaid in program B. the recipient user is enrolled in program B. should the sender have permission to email the recipient in this case? two things to consider: (1) a user pays for a course _run_, not a course or program, and (2) a learner sends a message to another learner from the profile page, where there is no context about the enrolled program.

i'm going ahead with the following behavior: a sender should be able to email a recipient if the sender has paid for any course run in any program that the recipient is enrolled in. let me know if you object to that
